File: ios/Sources/HttpLocalServerPlugin/HttpLocalServerPlugin.swift
// import Foundation
// import Capacitor

// /**
//  * Please read the Capacitor iOS Plugin Development Guide
//  * here: https://capacitorjs.com/docs/plugins/ios
//  */
// @objc(HttpLocalServerPlugin)
// public class HttpLocalServerPlugin: CAPPlugin, CAPBridgedPlugin {
//     public let identifier = "HttpLocalServerPlugin"
//     public let jsName = "HttpLocalServer"
//     public let pluginMethods: [CAPPluginMethod] = [
//         CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
//     ]
//     private let implementation = HttpLocalServer()

//     @objc func echo(_ call: CAPPluginCall) {
//         let value = call.getString("value") ?? ""
//         call.resolve([
//             "value": implementation.echo(value)
//         ])
//     }
// }

import Foundation
import Capacitor
@objc(HttpLocalServerPlugin)
public class HttpLocalServerPlugin: CAPPlugin, CAPBridgedPlugin, HttpLocalServerDelegate {
    public let identifier = "HttpLocalServerPlugin"
    public let jsName = "HttpLocalServer"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "connect", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "disconnect", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "sendResponse", returnType: CAPPluginReturnPromise)
    ]
    var localServer: HttpLocalServer?

    @objc func connect(_ call: CAPPluginCall) {
        if localServer == nil {
            localServer = HttpLocalServer(delegate: self)
        }
        localServer?.connect(call)
    }

    @objc func disconnect(_ call: CAPPluginCall) {
        if localServer != nil {
            localServer?.disconnect(call)
        } else {
            call.resolve()
        }
    }

    @objc func sendResponse(_ call: CAPPluginCall) {
        guard let requestId = call.getString("requestId"),
              let body = call.getString("body") else {
            call.reject("Missing requestId or body")
            return
        }
        HttpLocalServer.handleJsResponse(requestId: requestId, body: body)
        call.resolve()
    }

    // Delegate method
    public func httpLocalServerDidReceiveRequest(_ data: [String: Any]) {
        notifyListeners("onRequest", data: data)
    }
}

File: android/android/src/main/java/com/cappitolian/plugins/httplocalserver/HttpLocalServerPlugin.java
// package com.cappitolian.plugins.httplocalserver;

// import com.getcapacitor.JSObject;
// import com.getcapacitor.Plugin;
// import com.getcapacitor.PluginCall;
// import com.getcapacitor.PluginMethod;
// import com.getcapacitor.annotation.CapacitorPlugin;

// @CapacitorPlugin(name = "HttpLocalServer")
// public class HttpLocalServerPlugin extends Plugin {

//     private HttpLocalServer implementation = new HttpLocalServer();

//     @PluginMethod
//     public void echo(PluginCall call) {
//         String value = call.getString("value");

//         JSObject ret = new JSObject();
//         ret.put("value", implementation.echo(value));
//         call.resolve(ret);
//     }
// }

// package com.cappitolian.plugins.httplocalserver;

// import com.getcapacitor.JSObject;
// import com.getcapacitor.Plugin;
// import com.getcapacitor.PluginCall;
// import com.getcapacitor.PluginMethod;
// import com.getcapacitor.annotation.CapacitorPlugin;

// @CapacitorPlugin(name = "HttpLocalServer")
// public class HttpLocalServerPlugin extends Plugin {

//     private HttpLocalServer implementation = new HttpLocalServer();

//     @PluginMethod
//     public void echo(PluginCall call) {
//         String value = call.getString("value");

//         JSObject ret = new JSObject();
//         ret.put("value", implementation.echo(value));
//         call.resolve(ret);
//     }
// }

package com.cappitolian.plugins.httplocalservice;

import com.getcapacitor.*;
import com.getcapacitor.annotation.CapacitorPlugin;
import org.json.JSONException;
import org.json.JSONObject;

@CapacitorPlugin(name = "HttpLocalServer")
public class HttpLocalServerPlugin extends Plugin {

    private HttpLocalServer localServer;

    @PluginMethod
    public void connect(PluginCall call) {
        if (localServer == null) {
            localServer = new HttpLocalServer(this);
        }
        localServer.connect(call);
    }

    // Add this method:
    public void fireOnRequest(JSObject req) {
        notifyListeners("onRequest", req, true);
    }

    @PluginMethod
    public void disconnect(PluginCall call) {
        if (localServer != null) {
            localServer.disconnect(call);
        } else {
            call.resolve();
        }
    }

    @PluginMethod
    public void sendResponse(PluginCall call) {
        String requestId = call.getString("requestId");
        String body = call.getString("body");
        if (requestId == null || body == null) {
            call.reject("Missing requestId or body");
            return;
        }
        HttpLocalServer.handleJsResponse(requestId, body);
        call.resolve();
    }
}